staging/src/k8s.io/kubectl/pkg/cmd/apply/apply.go

```diff
@@ -407,11 +389,14 @@ func (o *ApplyOptions) Run() error {
 				Force:        &o.ForceConflicts,
 				FieldManager: o.FieldManager,
 			}
+
+			helper := resource.NewHelper(info.Client, info.Mapping)
 			if o.ServerDryRun {
-				options.DryRun = []string{metav1.DryRunAll}
+				if err := resource.VerifyDryRun(info.Mapping.GroupVersionKind, o.DynamicClient, o.DiscoveryClient); err != nil {
```

Comment:
Are we supposed to not print/fail on this error?

Reply:
Yeah, that's very dubious and might be the source of the broken test. If you have an error verifying if it's dry-runnable, then you're just going to apply non-dry-run, which should absolutely not happen.

Reply:
Yikes, thanks

Reply:
Fixed